bundle/Service/ScheduledVisibilityService.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Netgen\Bundle\IbexaScheduledVisibilityBundle\Service;
+
+use DateTime;
+use Ibexa\Contracts\Core\Repository\Values\Content\Content;
+use Ibexa\Contracts\Core\Repository\Values\Content\Field;
+use Ibexa\Core\FieldType\Date\Value;
+use Netgen\Bundle\IbexaScheduledVisibilityBundle\Enums\HandlerType;
+use Netgen\Bundle\IbexaScheduledVisibilityBundle\ScheduledVisibility\ScheduledVisibilityInterface;
+
+use function in_array;
+
+final class ScheduledVisibilityService
+{
+    public function __construct(
+        private readonly iterable $handlers,
+        private readonly string $type,
+        private readonly bool $enabled,
+        private readonly bool $allContentTypes,
+        private readonly array $allowedContentTypes,
+    ) {}
+
+    public function toggleVisibility(Content $content): void
```

Comment:
This method should throw `\OutOfBoundsException` if the handler can't be resolved.

Reply:
fixed in f603e3adab41a0d0e278a5683ddaa882923dd8ac